integration/emr/alluxio-emr.sh

```diff
@@ -10,99 +10,216 @@
+# Arg 5. (Optional) Delimeter for additional properties. Defaults to ;
+
+set -o errexit  # exit when a command fails - append "|| true" to allow a
+                # command to fail
+set -o nounset  # exit when attempting to use undeclared variables
+
+# Show commands being run - useful to keep by default so that failures
+# are easy to debug through AWS
+set -x
+
+# script constants
+ALLUXIO_HOME=/opt/alluxio
+ALLUXIO_SITE_PROPERTIES=${ALLUXIO_HOME}/conf/alluxio-site.properties
+
+# Downloads a file to the local machine from a remote HTTP(S) or S3 URI into the cwd
+#
+# Args:
+#   $1: URI - the remote location to retrieve the file from
+download_file() {
+  local uri=$1
+
+  if [[ "${uri}" == s3://* ]]
+  then
+    aws s3 cp ${uri} ./
+  else
+    # TODO Add metadata header tag to the wget for filtering out in download metrics.
+    wget -nv "${uri}"
+  fi
+}
+
+# Run a command as a specific user
+# Assumes the provided user already exists on the system and user running script has sudo access
+#
+# Args:
+#   $1: user - the username to run the command as
+#   $2: command - the command to run
+doas() {
+  local user=$1
+  local cmd=$2
+  sudo runuser -l ${user} -c "${cmd}"
+}
+
+# Appends a property KV pair to the alluxio-site.properties file
+#
+# Args:
+#   $1: property name
+#   $2: property value
+append_alluxio_property() {
+  local property=$1
+  local value=$2
+
+  # OK to fail in this section
+  set +o errexit
+  # ${ALLUXIO_SITE_PROPERTIES} must exist for this to work
+  grep -qe "^\s*${property}=" ${ALLUXIO_SITE_PROPERTIES} 2> /dev/null
+  local rv=$?
+  set -o errexit # errors not ok anymore
+  if [[ $rv -ne 0 ]]; then
+    doas alluxio "echo '${property}=${value}' >> ${ALLUXIO_SITE_PROPERTIES}"
+  fi
+}
+
+# Calculates the default memory size as 1/3 of the total system memory
+#
+# Echo's the result to stdout. To store the return value in a variable use
+# val=$(get_defaultmem_size)
+get_default_mem_size() {
+  local mem_div=3
+  local phy_total=$(free -m | grep -oP '\d+' | head -n1)
+  local mem_size=$(( ${phy_total} / ${mem_div} ))
+  echo "${mem_size}MB"
+}
+
+# Gets the region of the current EC2 instance
+get_aws_region() {
+  local ec2_avail_zone=$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)
+  local ec2_region="$(echo "${ec2_avail_zone}" | sed 's/[a-z]$//')"
+  echo "${ec2_region}"
+}
+
+main() {
+  local alluxio_tarball
+  local root_ufs_uri
+  local site_properties_uri
+  local property_delimiter
+  local delimited_properties
+  alluxio_tarball=${1}
+  root_ufs_uri=${2}
+  site_properties_uri=${3:-""}
+  delimited_properties=${4:-""}
+  property_delimiter=${5:-";"}
+
+  # Create user
+  sudo groupadd alluxio -g 600
+  sudo useradd alluxio -u 600 -g 600
+
+  # Download the release
+  if [[ -z ${alluxio_tarball} ]]
   then
-    aws s3 cp ${1} ./
+    echo "No Download URL Provided. Please go to http://downloads.alluxio.io to see available release downloads."
```

Comment:
exit here?

Reply:
Don't need to. The script specifies at the top 

```bash
set -o errexit  # exit when a command fails - append "|| true" to allow a
                # command to fail
```